File: Numerical_Methods/Numerical Solutions of Algebraic Equations/Regular-Falsi Method/python_solutions/problem6.py
import math

def f(x):
    return math.cos(x) - x

def regular_falsi(a, b, tolerance):
    if f(a) * f(b) >= 0:
        print("Initial points must have opposite signs")
        return None
    
    iteration = 0
    while True:
        # Calculate new root using Regular Falsi formula
        c = a - (f(a) * (b - a)) / (f(b) - f(a))
        
        # Print current iteration details
        print(f"Iteration {iteration}:")
        print(f"a = {a:.6f}, b = {b:.6f}, c = {c:.6f}")
        print(f"f(a) = {f(a):.6f}, f(b) = {f(b):.6f}, f(c) = {f(c):.6f}")
        print("-" * 50)
        
        # Check if we've reached the desired accuracy
        if abs(f(c)) < tolerance:
            return c
        
        # Update interval
        if f(c) * f(a) < 0:
            b = c
        else:
            a = c
            
        iteration += 1

# Initial interval and tolerance
a = 0
b = math.pi/2
tolerance = 0.0001

# Solve the equation
root = regular_falsi(a, b, tolerance)

if root is not None:
    print(f"\nFinal root: {root:.6f}")
    print(f"f(root) = {f(root):.6f}") 

File: Numerical_Methods/Solving Systems of Linear Equations/Gauss-Seidel Iterations/java_solutions/Problem6.java
public class Problem6 {
    public static void main(String[] args) {
        // System of equations:
        // 3x + 2y - z = 6
        // 2x + 4y + z = 7
        // -x + y + 5z = 8
        
        double x = 1.0, y = 1.0, z = 1.0; // Initial values
        double xNew, yNew, zNew;
        double tolerance = 0.001;
        int iteration = 0;
        boolean converged = false;

        System.out.println("Gauss-Seidel Iteration Method - Problem 6");
        System.out.println("Initial values: x = " + x + ", y = " + y + ", z = " + z);
        System.out.println("Tolerance: " + tolerance);
        System.out.println("\nIterations:");

        while (!converged) {
            iteration++;
            
            // Calculate new values using Gauss-Seidel formula
            xNew = (6 - 2*y + z) / 3;
            yNew = (7 - 2*xNew - z) / 4;
            zNew = (8 + xNew - yNew) / 5;

            // Check for convergence
            if (Math.abs(xNew - x) < tolerance &&
                Math.abs(yNew - y) < tolerance &&
                Math.abs(zNew - z) < tolerance) {
                converged = true;
            }

            // Update values
            x = xNew;
            y = yNew;
            z = zNew;

            // Print current iteration
            System.out.printf("Iteration %d: x = %.6f, y = %.6f, z = %.6f%n", 
                            iteration, x, y, z);
        }

        System.out.println("\nSolution converged after " + iteration + " iterations:");
        System.out.printf("x = %.6f%n", x);
        System.out.printf("y = %.6f%n", y);
        System.out.printf("z = %.6f%n", z);
    }
} 